packages/contracts-bedrock/test/vendor/Initializable.t.sol

```diff
@@ -225,242 +235,273 @@ contract Initializer_Test is Bridge_Initializer {
-    ///      For L1 contracts, implementations are considered in addition to proxies
-    function _getNumInitializable() internal returns (uint256 numContracts_) {
-        string[] memory command = new string[](3);
-        command[0] = Executables.bash;
-        command[1] = "-c";
-        // Start by getting L1 contracts
-        command[2] = string.concat(
-            Executables.find,
-            " src/L1 -type f -exec basename {} \\;",
-            " | ",
-            Executables.sed,
-            " 's/\\.[^.]*$//'",
-            " | ",
-            Executables.jq,
-            " -R -s 'split(\"\n\")[:-1]'"
-        );
-        string[] memory l1ContractNames = abi.decode(vm.parseJson(string(Process.run(command))), (string[]));
-
-        for (uint256 i; i < l1ContractNames.length; i++) {
-            string memory contractName = l1ContractNames[i];
-            string memory contractAbi = ForgeArtifacts.getAbi(contractName);
+            // Filter out non-contracts.
+            if (!LibString.eq(contractKind, "contract")) {
+                continue;
+            }
 
-            // Query the contract's ABI for an `initialize()` function.
+            // Construct the query for the initialize function in the contract's ABI.
+            string[] memory command = new string[](3);
+            command[0] = Executables.bash;
+            command[1] = "-c";
             command[2] = string.concat(
                 Executables.echo,
                 " '",
-                contractAbi,
+                ForgeArtifacts.getAbi(contractName),
                 "'",
                 " | ",
                 Executables.jq,
                 " '.[] | select(.name == \"initialize\" and .type == \"function\")'"
             );
-            bytes memory res = Process.run(command);
 
-            // If the contract has an `initialize()` function, the resulting query will be non-empty.
-            // In this case, increment the number of `Initializable` contracts.
-            if (res.length > 0) {
-                // Count Proxy + Impl
-                numContracts_ += 2;
+            // If the contract does not have an `initialize()` function, skip it.
+            if (Process.run(command).length == 0) {
+                continue;
+            }
+
+            // Check if this contract is in the contracts array.
+            assertTrue(
+                _hasMatchingContract(contractName), string.concat("Missing ", contractName, " from contracts array")
+            );
+
+            // If the contract is proxied, check that the proxy is in the contracts array.
+            // Skip predeployed contracts for now since we don't yet keep track of the
+            // implementations inside of the deploy script. We should add support for this in the
+            // future so that we can properly check that the implementations for predeployed
+            // contracts are initialized too.
```

Comment:
Let's add a `TODO` athe start of the of the comment here so this todo is more easily discoverable